### PR TITLE
adding permissions to the NFD controller

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
adding kubebuiler options for creating RBAC permissions needed for NFD controller, and creating the actula role yaml

### CODE REVIEW

File: internal/controllers/nodefeaturediscovery_reconciler.go [hunk 1]
```diff
@@ -99,7 +99,11 @@ func isControlledByNFD(obj client.Object) bool {
 // +kubebuilder:rbac:groups=apps,resources=daemonsets,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=core,resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;watch;create;update;patch;delete
```
Comment (ArangoGutierrez) [inline internal/controllers/nodefeaturediscovery_reconciler.go:102]: Batch?

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
Why do we need RBAC for `batch`?

File: internal/controllers/nodefeaturediscovery_reconciler.go (hunk 1 above)
Comment (yevgeny-shnaidman) [reply internal/controllers/nodefeaturediscovery_reconciler.go:102]: we are handing prune Jobs

Comment (ArangoGutierrez) [reply internal/controllers/nodefeaturediscovery_reconciler.go:102]: ah the prune stuff..... my bad

**Review by ArangoGutierrez — APPROVED** [review]
/lgtm

**Pull request merged**